defender/wallet-migrator/README.md

```diff
@@ -0,0 +1,72 @@
+
+## Defender Account Setup
+
+- In your [Defender account](https://defender.openzeppelin.com/), select the Hamburger icon in the upper right corner and click on **Team API Keys**
+- In the Team API Keys screen, click **Create API Key**
+- Make sure that the options for **Manage Relayers**, **Manage Autotasks**, and **Manage Sentinels** are selected (we do not make use of the **Manage Proposals and Contracts** option)
+- Click **Save**
+- Copy your API key and Secret key to a local file (you will **NOT** be able to view your API secret again after this message box goes away)
+- Make sure that you really did copy your API key and Secret key to a local file
+- Check the box for **I’ve written down the secret key** and select **Close**
+
+### Covalent API Key Setup
+
+This template will make use of [Covalent's API](https://www.covalenthq.com/docs/api/balances/get-token-balances-for-address/) to query for all 
+all balances of ERC20 and ERC721 tokens of the sender's specified wallet address. To get an API key, navigate to [Covalent's website](https://www.covalenthq.com/)
+and select **Get an API key**. Follow the instructions to sign up for an account, or login to your existing account if you already have one. 
+Once you are logged in and have created an API Key, you can navigate to your home platform [page](https://www.covalenthq.com/platform/#/) where the API key is shown. 
+Write down your API Key is a safe place, as you will need it for later steps.
+
+## Local Code Setup
+
+In the `defender` directory, perform the following steps:
+
+- Run `yarn install` to install the necessary Node packages
+- A secrets file can be created for each stage of production. We will create one for development
+  - Copy and rename `sample.secrets.yml` to `.secrets/dev.yml`
+  - Modify the following lines in the `.secrets/dev.yml` file, replacing the portion in the angle brackets `<>` with your Defender API key, Secret key, and Covalent API key, as indicated:
+    - `defender-api-key: <API Key goes here>`
+    - `defender-api-secret: <API Secret goes here>`
+    - `covalent-api-key: <Covalent API Key goes here>`
+- Change directories to the stack that will be deployed
+  - `cd wallet-migrator`
+- A config file can be created for each stage of production. We will create one for development
+  - Copy and rename `sample.config.yml` to `config.dev.yml`
+  - Open the file and modify the following values:
+    - `sender-wallet-address` 
+    - `receiver-wallet-address`
+    - `network`
+- Run `serverless deploy --stage dev` to deploy the stack to Defender
+
+## Token Approval Script Setup
+
+In order to have all ERC20 and ERC721 tokens automatically transferred, you must first approve an allowance to the Relayer that is deployed with your Defender Stack.
+
+From the root of this repo, follow these steps to run the approval script:
+
+- Create/modify the `.env` file
+- Add your RPC URL to perform on-chain transactions
```

Comment:
Take or leave it: The newly deployed relayer could act as the provider. The relayer key and secret would need to be added to `secrets.yml` after the `serverless deploy`

Sample code:
```js
  const req = require('require-yml');
  const secrets = req(`../.secrets/${stage}.yml`);
  const {
	'relay-api-key': relayApiKey,
	'relay-api-secret': relaySecretKey,
  } = secrets.keys;

  if (relayApiKey === undefined) {
	throw new Error('Could not find Relay API key in defender/.secrets/<stage>.yml');
  }
  if (relaySecretKey === undefined) {
	throw new Error('Could not find Relay Secret key in defender/.secrets/<stage>.yml');
  }

  // Test Defender Relay
  const credentials = { apiKey: relayApiKey, apiSecret: relaySecretKey };
  const provider = new DefenderRelayProvider(credentials);
  const signer = new DefenderRelaySigner(credentials, provider, {
	speed: 'fast',
  });
  try {
    await signer.getAddress();
  } catch (error) {
    throw new Error(`Issue with Defender Relay: ${error}`);
  }
  ```

Reply:
I ended up taking gutavo's suggestion and running the approve tokens script with hardhat. Because of that, I was able to grab the rpc endpoint and the chain id from hardhat config and this simplified it